packages/components/src/internal/components/forms/input/UserSelectInput.tsx

```diff
@@ -29,13 +35,73 @@ function generateKey(permissions?: string | string[], containerPath?: string): s
 
 interface UserSelectInputProps extends Omit<SelectInputProps, 'delimiter' | 'loadOptions'> {
     containerPath?: string;
+    includeGroups?: boolean;
     includeInactive?: boolean;
+    includeSiteGroups?: boolean;
     // specify whether this Select should correspond with a NotifyList on the server
     notifyList?: boolean;
     permissions?: string | string[];
     useEmail?: boolean;
 }
 
+export const getUserGroupOptions = (
+    users: User[],
+    groups?: FetchedGroup[],
+    input?: string,
+    notifyList?: boolean,
+    useEmail?: boolean,
+    includeSiteGroups?: boolean
+): SelectInputOption[] => {
+    let userOptions: SelectInputOption[];
+    const sanitizedInput = input?.trim().toLowerCase();
+    userOptions = users
+        ?.filter(v => {
+            if (sanitizedInput) {
+                return v.displayName?.toLowerCase().indexOf(sanitizedInput) > -1;
+            }
+
+            return true;
+        })
+        .map(v => ({
+            label: v.displayName,
+            value: notifyList ? v.displayName : useEmail ? v.email : v.userId,
+        }));
+    if (groups?.length > 0) {
+        const groupedUserOptions = {
+            label: 'Users',
+            options: userOptions,
+        };
+        const groupOptions = groups
+            .filter(group => {
+                if (sanitizedInput && group.name?.toLowerCase().indexOf(sanitizedInput) === -1) {
+                    return false;
+                }
+
+                if (includeSiteGroups) return true;
+                return group.isProjectGroup;
+            })
+            .map(v => ({
+                label: v.name,
+                value: v.id,
+            }));
+        if (groupOptions?.length > 0) {
+            const groupedGroupOptions = {
+                label: 'Project Groups',
```

Comment:
seems like if we are going to keep the `includeSiteGroups` option, then this needs to be conditional. If `includeSiteGroups` is never true, then I'd suggest we just drop that param for now.

Reply:
Nope `includeSiteGroups` is not currently used. Removed.